# Balance Sheet Efficiency Ratios - Part 2

**Lesson:** Balance sheet and forecast ratios  
_Financial Forecasting in Python_

📄 Notes: [03 - Balance Sheet Efficiency Ratios - Part 2](docs/03%20-%20Balance%20Sheet%20Efficiency%20Ratios%20-%20Part%202.md)

## Overview

Calculate the second set of balance-sheet efficiency ratios, building on Part 1.

## Learning objectives

- TODO: objective 1
- TODO: objective 2
- TODO: objective 3

## Setup

In [7]:
import sys; sys.path.insert(0, '../../src')

# import numpy as np
import pandas as pd
# import matplotlib.pyplot as plt
#
from ffp import sample_prices, error_report  # shared helpers

## Walkthrough

In [8]:
# TODO: lesson code here
# prices = sample_prices()
# prices.plot(figsize=(10, 4), title='Sample price series')
# plt.show()

**Exercise 1 - Days in inventory and asset turnover ratio**

In this exercise, we will calculate the time it takes for a company to turn inventory into sales (days in inventory or DII ratio) and a ratio to calculate the efficiency of a company’s assets by seeing how the company uses its assets to generate sales (asset turnover ratio). The required formulas are:

$$\text{Days in Inventory} = \left( \frac{\text{Average Inventory}}{\text{Total COGS}} \right) \times \text{Days in Financial Year}$$

$$\text{Total Average Assets} = \frac{\text{Opening Balance} + \text{Closing Balance}}{2}$$

$$\text{Asset Turnover Ratio} = \frac{\text{Sales}}{\text{Total Average Assets}}$$

The following values have been defined for you:

| Metric                   | Variable   | Value |
|--------------------------|------------|-------|
| Total COGS               | cogs_tot   | 4000  |
| Average Inventory        | av_inv     | 1900  |
| Total Sales              | sales_tot  | 10000 |
| Opening balance Assets   | ob_assets  | 2000  |
| Closing balance Assets   | cb_assets  | 7000  |

In [9]:
cogs_tot  = 4000
av_inv     = 1900
sales_tot  = 10000
ob_assets  = 2000
cb_assets  = 7000
# # Calculate the dii ratio
dii_ratio = (av_inv/cogs_tot)*365

# Print the result
print("The DII ratio is {}.".format(dii_ratio))

# Calculate the Average Assets
av_assets = (ob_assets + cb_assets)/2

# Calculate the Asset Turnover Ratio
at_ratio = sales_tot/av_assets

# Print the Asset Turnover Ratio
print("The asset turnover ratio is {}.".format(at_ratio))

The DII ratio is 173.375.
The asset turnover ratio is 2.2222222222222223.


**Exercise 2 - Understanding ratios**

T-Z has decided to do a comparison and analysis between their company, and a competitor, ShirtCo. Below is the information collected:

| Ratio                    | T-Z  | ShirtCo |
|--------------------------|------|---------|
| Debtor Days Ratio        | 17.5 | 6.8     |
| Days Payable Outstanding | 59   | 91      |
| Inventory Days Ratio     | 171  | 205     |
| Asset Turnover           | 1.1  | 9.5     |

**What conclusion could T-Z come to based on this comparison?**

_T-Z takes longer to collect cash from their debtors, but is more efficient in managing their inventory than ShirtCo._

T-Z manages their creditors better than ShirtCo.

ShirtCo is more efficient to collect cash from debtors but needs a higher investment in assets than T-Z.

ShirtCo requires a lower investment in assets than T-Z and also has lower days in inventory than T-Z.


**Exercise 3 - Calculating balance sheet ratios for Ford**

Now we will look at a real life example, Ford Inc, a company producing motor vehicles. We have uploaded a dataset: balance_sheet with the data for Ford Inc's most recent balance sheet. The sales and cost of sales figures have been provided for 2017 with the Key_Figures_Memo dataset.

We are only interested in one line on the balance sheet, the Receivables (another name for Debtors), and therefore need to create a filter for this. In this exercise, we will use boolean indexing to filter our dataset for Receivables in the metric column. We will first specify our metric of interest ('Receivables'), and then check whether the column of interest has this value in each row. This will generate a boolean series of True and False values. With this series, we can then filter our existing dataset.

Once we have filtered our dataset, we can retrieve the receivables values from the most recent time period and calculate the debtor days ratio (provided below).

$$\text{Debtor Days} = \left( \frac{\text{Ending Balance Debtors}}{\text{Sales}} \right) \times \text{Days in Financial Year}$$

The `balance_sheet` and `sales` figures are provided.



In [11]:
balance_sheet = pd.read_csv('../../data/ford_balance_sheet.csv', skiprows=1)
balance_sheet = balance_sheet.rename(columns={balance_sheet.columns[0]: 'metric'})

sales = 156776
# Create the filter metric for Receivables
receivables_metric = ['Receivables']

# Create a boolean series with your metric
receivables_filter = balance_sheet.metric.isin(receivables_metric)

# Use the series to filter the dataset
filtered_balance_sheet = balance_sheet[receivables_filter]

# Extract the zeroth value from the last time period (2017-12)
debtors_end = filtered_balance_sheet['2017-12'].iloc[0]

# Calculate the debtor days ratio
ddays = (debtors_end / sales) * 365

# Print the debtor days ratio
print("The debtor day ratio is {:.0f}. A higher debtors days ratio means it takes longer to collect cash from debtors.".format(ddays))

The debtor day ratio is 146. A higher debtors days ratio means it takes longer to collect cash from debtors.


**Exercise 4 - Forecasting the balance sheet for Ford**

Now that we have the ratios for Ford, the management wishes to improve them for the 2018 financial year.

There is one ratio in particular that management would like to be improved – the debtor days ratio, which is currently sitting at 146 days, while better than Nissan (244 days), is still much higher than Toyota, the industry leader (31 days).

Management would like to bring the debtor days down to below 100 days.

These variables have already been set:

filtered_balance_sheet contains the receivables row of the balance sheet.
* ddays = 99 is the target debtor days.
* sales = 156776 is the existing sales total.

The formula to calculate debtor days is:
$$\text{Debtor Days} = \left( \frac{\text{Closing Balance Debtors}}{\text{Sales}} \right) \times \text{Days in Financial Year}$$

In [12]:
ddays = 99
sales = 156776
# Calculate the forecasted sales
f_sales = sales * 1.10

# Solve for the forecasted debtors' ending balance
f_debtors_end = f_sales * ddays / 365

print("If sales rise by 10% and the debtor days decrease to {:.0f} then the forecasted closing balance for debtors will be {:.0f}.".format(ddays, f_debtors_end))

# Get the number of columns in the filtered balance sheet
n_cols = len(filtered_balance_sheet.columns)

# Append a Forecast column of the forecasted debtors' end balance
filtered_balance_sheet.insert(n_cols, 'Forecast', f_debtors_end)

# See the result
print(filtered_balance_sheet)

If sales rise by 10% and the debtor days decrease to 99 then the forecasted closing balance for debtors will be 46775.
        metric  2013-12  2014-12   2015-12  2016-12  2017-12      Forecast
6  Receivables  87309.0  92819.0  101975.0  57368.0  62809.0  46775.086027


## Summary

Recap the key points in the notes: [03 - Balance Sheet Efficiency Ratios - Part 2](docs/03%20-%20Balance%20Sheet%20Efficiency%20Ratios%20-%20Part%202.md)